src/scripts/import-multiple-files.ts

```diff
@@ -0,0 +1,102 @@
+import { readdir, readFile, writeFile } from "node:fs/promises";
+
+import { D2Api } from "./../types/d2-api";
+import appConfig from "../../public/app-config.json";
+import { getCompositionRoot } from "../CompositionRoot";
+import { JsonConfig } from "../data/ConfigWebRepository";
+import { getD2APiFromInstance } from "../utils/d2-api";
+import Settings from "../webapp/logic/settings";
+import { promiseMap } from "../utils/promises";
+import { isExcelFile } from "../utils/files";
+
+type BufferWithFileName = {
+    content: Buffer;
+    fileName: string;
+};
+
+async function readExcelFilesInFolder(fullPathFolder: string): Promise<BufferWithFileName[]> {
+    const fileNames = await readdir(fullPathFolder);
+    console.debug(`Looking for excel files in: ${fullPathFolder}`);
+    const filesContent = await promiseMap(fileNames, async fileName => {
+        const filePath = resolve(fullPathFolder, fileName);
+        const content = isExcelFile(fileName) ? await readFile(filePath) : undefined;
+        if (!content) return undefined;
+        return {
+            content,
+            fileName,
+        };
+    });
+    const excelFiles = _(filesContent).compact().value();
+    console.debug(`${excelFiles.length} excel files found.`);
+    return excelFiles;
+}
+
+function main() {
+    const cmd = command({
+        name: path.basename(__filename),
+        description: "Import data from multiples excel files",
+        args: {
+            url: option({
+                type: string,
+                short: "u",
+                long: "dhis2-url",
+                description: "DHIS2 base URL. Example: http://USERNAME:PASSWORD@localhost:8080",
+            }),
+            folderPath: option({
+                type: string,
+                short: "fp",
+                long: "folder-path",
+                description: "folder path with excel files to import data",
+            }),
+            resultsPath: option({
+                type: string,
+                short: "rp",
+                long: "results-path",
+                description: "folder where import results will be saved",
+            }),
+        },
+        handler: async args => {
+            const filesContent = await readExcelFilesInFolder(args.folderPath);
+            const api: D2Api = getD2APiFromInstance({ type: "local", url: args.url });
+
+            await promiseMap(filesContent, async file => {
+                const compositionRoot = getCompositionRoot({
+                    appConfig: appConfig as unknown as JsonConfig,
+                    dhisInstance: { type: "local", url: args.url },
+                    importSource: "node",
+                });
+                const settings = await Settings.build(api, compositionRoot);
+                console.debug(`Importing file ${file.fileName}`);
+                const results = await compositionRoot.templates.import({
+                    // @ts-ignore
+                    file: file.content,
```

Comment:
The solution on the long term for that is having a FileEntity instead a concrete File. The repos can build them from a browser/node accordingly. Maybe here it can be as simple as `File | Buffer`. For the next iteration.